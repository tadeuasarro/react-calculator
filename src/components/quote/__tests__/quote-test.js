import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Quote from '../quote';

describe('Quote', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<BrowserRouter><Quote /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('creates a Quote component', () => {
    render(<BrowserRouter><Quote /></BrowserRouter>);
    const element = screen.getByText('BALBOA, Rocky');
    expect(element).toBeInTheDocument();
  });
});
