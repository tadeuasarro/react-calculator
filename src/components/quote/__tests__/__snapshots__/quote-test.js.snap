// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote renders correctly 1`] = `
<main>
  <div
    id="navbar"
  >
    <h1>
      Math-magicians
    </h1>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/app"
          onClick={[Function]}
        >
          App
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </div>
  <div
    id="outer-container"
  >
    <div
      id="quote"
    >
      <p>
        Let me tell you something you already know. The world ain't all sunshine and rainbows. It's a very mean and nasty place and I don't care how tough you are it will beat you to your knees and keep you there permanently if you let it. You, me, or nobody is gonna hit as hard as life. But it ain't about how hard ya hit. It's about how hard you can get hit and keep moving forward. How much you can take and keep moving forward. That's how winning is done!
      </p>
      <p
        id="author"
      >
        BALBOA, Rocky
      </p>
    </div>
  </div>
</main>
`;
